File: java/PulsanteLed.java
import com.pi4j.io.gpio.GpioController;
import com.pi4j.io.gpio.GpioFactory;
import com.pi4j.io.gpio.GpioPinDigitalOutput;
import com.pi4j.io.gpio.PinState;
import com.pi4j.io.gpio.RaspiPin;
import com.pi4j.io.gpio.GpioPinDigitalInput;
import com.pi4j.io.gpio.PinPullResistance;
import com.pi4j.io.gpio.event.GpioPinDigitalStateChangeEvent;
import com.pi4j.io.gpio.event.GpioPinListenerDigital;

public class PulsanteLed {

    public static void main(String args[]) throws InterruptedException {
        System.out.println("Lampeggio LED GPIO 12");
        
        //creo il controller
        final GpioController gpio = GpioFactory.getInstance();

        //imposto myButton come input dal pin 0 (GPIO11) e imposto pin come output sul pin 1 (GPIO12)
        final GpioPinDigitalInput myButton = gpio.provisionDigitalInputPin(RaspiPin.GPIO_00, PinPullResistance.PULL_DOWN);
       	final GpioPinDigitalOutput pin = gpio.provisionDigitalOutputPin(RaspiPin.GPIO_01, "MyLED", PinState.LOW);

        //mi metto in ascolto dello stato del pin in input (listener)
        myButton.addListener(new GpioPinListenerDigital() {
            @Override
            public void handleGpioPinDigitalStateChangeEvent(GpioPinDigitalStateChangeEvent event) {

                System.out.println("PULSANTE: " + event.getState());
		pin.toggle(); //cambio di stato il LED
        	System.out.println("LED ACCESO");
		try {
 			Thread.sleep(1000); //attendo 1 sec.
		}
		catch (InterruptedException ie) {
 			//gestisci eccezzione
		}
            }
            
        });
        
        //creo un loop infinito solo per attendere l'evento (pulsante premuto)
        for (;;) {
            Thread.sleep(500);
        }
    }
}

File: c/PulsanteLed.c
#include <stdio.h>
#include <wiringPi.h>

//vedi tabella pin della GPIO, colonna wiringPi
#define LED12 1 //alla variabile LED12 assegno il pin 0
#define PULSANTE11 0 //alla variabile PULSANTE11 assegno il pin 1

int main(void)
{
	printf("Lampeggio LED GPIO 12\n");

	wiringPiSetup(); //inizializzo la libreria

	pinMode(LED12,OUTPUT); //essendo un LED lo imposto come "output"
	pinMode(PULSANTE11,INPUT); //essendo un pulsante lo imposto come "input"

	for (;;) //ciclo infinito
	{
		if (digitalRead(PULSANTE11)==LOW) //premuto il pulsante
		{
			printf("LED ACCESO\n");
			digitalWrite(LED12,HIGH); //accendo il LED
			delay(1000); //acceso per 1 sec.
		}
		printf("LED spento\n");
		digitalWrite(LED12,LOW); //spengo il LED
		delay(1000); //rallento un po'
	}
return 0; //chiudo main
}

